packages/server/api/src/app/workers/queue/queue-events/save-queue-metrics.ts

```diff
@@ -0,0 +1,80 @@
+import { FastifyBaseLogger } from "fastify";
+import { Job, Queue, QueueEvents } from "bullmq";
+import { EventsManager } from "./events-manager";
+import { redisConnections } from "../../../database/redis";
+import { JobData, WorkerJobStatus, WorkerJobType } from "@activepieces/shared";
+import { QueueName } from "@activepieces/server-shared";
+import { bullMqGroups } from "../job-queue";
+import { Redis } from "ioredis";
+
+export const METRICS_KEY = "metrics"
+export const JOB_STATE_KEY = "job_state"
+
+export const METRICS_KEY_PREFIX = (jobType: WorkerJobType) => `${METRICS_KEY}:${jobType}`
+export const JOB_STATE_KEY_PREFIX = (jobId: string) => `${JOB_STATE_KEY}:${jobId}`
```

Comment:
This shoudl have workerJobStatus, `metricsKeyForJobType` and the other one would be `jobStateKey`